### PR TITLE
Support Stdin and TTY in the kubelet

### PR DESCRIPTION
**Pull request by brendandburns**
@dchen1107 

This depends on #12007

### CODE REVIEW

**Comment by brendandburns** [comment]
Hold on this...

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -588,11 +588,13 @@ func (dm *DockerManager) runContainer(
...
-			MemorySwap: -1,
-			CPUShares:  cpuShares,
-			WorkingDir: container.WorkingDir,
-			Labels:     labels,
+			Memory:      memoryLimit,
+			MemorySwap:  -1,
+			CPUShares:   cpuShares,
+			WorkingDir:  container.WorkingDir,
+			Labels:      labels,
+			AttachStdin: container.Stdin,
```
Comment (bgrant0607) [inline pkg/kubelet/dockertools/manager.go]: @dchen1107 What's our current policy on feature parity for Rocket?

Comment (brendandburns) [inline pkg/kubelet/dockertools/manager.go]: I think it has to be best effort.  I'm happy to log an event or something to indicate the error.

**Comment by brendandburns** [comment]
Ok, validated that this works correctly, this is ready to submit.

**Comment by pmorie** [comment]
@ncdc

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/dockertools/manager.go]: Yes, it is a best effort at this moment. Even for today's rkt support, there are several features are missing / underdevelopment, for example, kubectl logs is missing from rkt. 

I am thinking about letting kubelet maintain a small list of feature set for container runtime version, and publish it as NodeStatus?

**Comment by dchen1107** [comment]
cc/ @yifan-gu on rkt side.

Can we have a simple e2e test for this? I am ok with a separate pr though.

LGTM

**Comment by yifan-gu** [comment]
@dchen1107 Thanks! Tracked in #8262

**Comment by brendandburns** [comment]
Yes, will add an e2e (not sure how easy it will be to write ;)

**Comment by dchen1107** [comment]
@brendandburns both shippable and travis are failed due to:

!!! 'gofmt -s' needs to be run on the following files:
./pkg/kubelet/dockertools/manager.go

**Comment by brendandburns** [comment]
@dchen1107 fixed gofmt, sorry about that.

**Comment by dchen1107** [comment]
LGTM

**Comment by wojtek-t** [comment]
@brendanburns @dchen1107 

Sorry - merging this PR broke Jenkins/Travis/Shippable. It seems it's because not-synces with change "github.com/GoogleCloudPlatform" => "k8s.io".
I'm reverting this one - please resend it after fixing that.

**Pull request merged**